govcd/edgegateway_lb_service_monitor.go

```diff
@@ -0,0 +1,187 @@
+	lbMonitorID := splitLocation[len(splitLocation)-1]
+	readMonitor, err := eGW.ReadLBServiceMonitor(&types.LBMonitor{ID: lbMonitorID})
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve monitor with ID (%s) after creation: %s", readMonitor.ID, err)
+	}
+	return readMonitor, nil
+}
+
+// ReadLBServiceMonitor is able to find the types.LBMonitor type by Name and/or ID.
+// If both - Name and ID are specified it performs a lookup by ID and returns an error if the specified name and found
+// name do not match.
+func (eGW *EdgeGateway) ReadLBServiceMonitor(lbMonitorConfig *types.LBMonitor) (*types.LBMonitor, error) {
+	if err := validateReadLBServiceMonitor(lbMonitorConfig); err != nil {
+		return nil, err
+	}
+
+	httpPath, err := eGW.getProxiedEdgeEndpoint("/loadbalancer/config/monitors")
+	if err != nil {
+		return nil, fmt.Errorf("could not get Edge Gateway API endpoint: %s", err)
+	}
+
+	// Anonymous struct to unwrap "monitor response"
+	lbMonitorResponse := &struct {
+		LBMonitors []*types.LBMonitor `xml:"monitor"`
+	}{}
+
+	// This query returns all service monitors as the API does not have filtering options
+	_, err = eGW.client.ExecuteRequest(httpPath, http.MethodGet, "application/xml", "unable to read Load Balancer monitor: %s", nil, lbMonitorResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	// Search for monitor by ID or by Name
+	for _, monitor := range lbMonitorResponse.LBMonitors {
+		// If ID was specified for lookup - look for the same ID
+		if lbMonitorConfig.ID != "" && monitor.ID == lbMonitorConfig.ID {
+			return monitor, nil
+		}
+
+		// If Name was specified for lookup - look for the same Name
+		if lbMonitorConfig.Name != "" && monitor.Name == lbMonitorConfig.Name {
+			// We found it by name. Let's verify if search ID was specified and it matches the lookup object
+			if lbMonitorConfig.ID != "" && monitor.ID != lbMonitorConfig.ID {
+				return nil, fmt.Errorf("load balancer monitor was found by name (%s), but it's ID (%s) does not match specified ID (%s)",
+					monitor.Name, monitor.ID, lbMonitorConfig.ID)
+			}
+			return monitor, nil
+		}
+	}
+
+	return nil, fmt.Errorf("could not find load balancer service monitor (name: %s, ID: %s)",
+		lbMonitorConfig.ID, lbMonitorConfig.Name)
+}
+
+// UpdateLBServiceMonitor
+func (eGW *EdgeGateway) UpdateLBServiceMonitor(lbMonitorConfig *types.LBMonitor) (*types.LBMonitor, error) {
+	if err := validateUpdateLBServiceMonitor(lbMonitorConfig); err != nil {
+		return nil, err
+	}
+
+	// if only name was specified for update, ID must be found, because ID is mandatory for update
+	if lbMonitorConfig.ID == "" {
+		readLBMonitor, err := eGW.ReadLBServiceMonitor(&types.LBMonitor{Name: lbMonitorConfig.Name})
+		if err != nil {
+			return nil, fmt.Errorf("unable to find load balancer monitor by name for update: %s", err)
+		}
+		lbMonitorConfig.ID = readLBMonitor.ID
+	}
+
+	httpPath, err := eGW.getProxiedEdgeEndpoint("/loadbalancer/config/monitors/" + lbMonitorConfig.ID)
```

Comment:
`/loadbalancer/config/monitors/` can be constant

Reply:
Moved to constants.